cmd/clusterctl/cmd/init.go

```diff
@@ -56,11 +56,12 @@ var initCmd = &cobra.Command{
 		The management cluster must be an existing Kubernetes cluster, make sure
 		to have enough privileges to install the desired components.
 
-		Use 'clusterctl config repositories' to get a list of available providers; if necessary, edit
-		$XDG_CONFIG_HOME/cluster-api/clusterctl.yaml file to add new provider or to customize existing ones.
+		Use 'clusterctl config repositories' to get a list of available providers and their repository configuration; if
+		necessary, edit $XDG_CONFIG_HOME/cluster-api/clusterctl.yaml file to add new provider or to customize existing ones.
 
 		Some providers require environment variables to be set before running clusterctl init.
-		Refer to the provider documentation, or use 'clusterctl config provider [name]' to get a list of required variables.
+		Refer to the provider documentation, or use 'clusterctl generate cluster [name] --list-variables --infrastructure [name]'
```

Comment:
Sorry my bad. Should have thought about the suggestion in the issue :D 

The correct command here is 'clusterctl generate provider --infrastructure [name] --describe'

e.g.

```bash
 clusterctl generate provider --infrastructure aws --describe
Name:               aws
Type:               InfrastructureProvider
URL:                https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/
Version:            v2.3.1
File:               infrastructure-components.yaml
TargetNamespace:    capa-system
Variables:
  - AUTO_CONTROLLER_IDENTITY_CREATOR
  - AWS_B64ENCODED_CREDENTIALS
  - AWS_CONTROLLER_IAM_ROLE
  - CAPA_EKS
  - CAPA_EKS_ADD_ROLES
  - CAPA_EKS_IAM
  - CAPA_LOGLEVEL
  - EVENT_BRIDGE_INSTANCE_STATE
  - EXP_ALTERNATIVE_GC_STRATEGY
  - EXP_BOOTSTRAP_FORMAT_IGNITION
  - EXP_EKS_FARGATE
  - EXP_EXTERNAL_RESOURCE_GC
  - EXP_MACHINE_POOL
  - K8S_CP_LABEL
  - TAG_UNMANAGED_NETWORK_RESOURCES
Images:
  - registry.k8s.io/cluster-api-aws/cluster-api-aws-controller:v2.3.1
```